### PR TITLE
PAYEE-249: Use java.time rather than joda time

### PR DESCRIPTION
**Pull request by sergi-hmrc**
- [x ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: test/data/RTIData.scala [hunk 1]
```diff
@@ -60,10 +60,10 @@ object RTIData {
       employment.copy(payments = employment.payments.map { inYear =>
         //We can finally try to change the value for the payment data
         inYear.copy(
-          paidOn = new LocalDate(
+          paidOn = LocalDate.of(
             TaxYear().year,
-            inYear.paidOn.getMonthOfYear,
-            inYear.paidOn.getDayOfWeek
+            inYear.paidOn.getMonth.getValue,
```
Comment (brendanstraughan) [inline test/data/RTIData.scala:65]: picky but can we use getMonthValue and getDayOfWeekValue?

File: app/uk/gov/hmrc/tai/model/templates/PdfSubmission.scala [hunk 2]
```diff
@@ -17,18 +17,16 @@
...
+import java.time.format.DateTimeFormatter
 import scala.util.Random
 
 case class PdfSubmission(
   customerId: String, //Needs to be the nino without suffix as DMS doesn't use suffix
   formId: String,
   numberOfPages: Int,
   attachmentCount: Int = 0,
-  hmrcReceivedAt: LocalDateTime = LocalDateTime.now(),
+  hmrcReceivedAt: LocalDateTime = LocalDateTime.now(ZoneId.of("Europe/London")), //TODO extract from config
```
Comment (brendanstraughan) [inline app/uk/gov/hmrc/tai/model/templates/PdfSubmission.scala:29]: has a ticket been raised for this? If so can we merge this before that ticket has been done?

Comment (sergi-hmrc) [reply app/uk/gov/hmrc/tai/model/templates/PdfSubmission.scala:29]: Yes: PAYEE-255

File: test/data/RTIData.scala (hunk 1 above)
Comment (sergi-hmrc) [reply test/data/RTIData.scala:65]: Sorry totally missed this one but changed another one

Comment (sergi-hmrc) [reply test/data/RTIData.scala:65]: getDayOfWeekValue doesn't exist

**Review by brendanstraughan — APPROVED** [review]

**Pull request merged**